compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -497,175 +484,93 @@ LogicalResult circularDmaToAIE(IRRewriter &rewriter,
     }
-    bool issueToken = dmaOp.hasDmaWaitOpUser();
-    rewriter.create<AIEX::NpuDmaMemcpyNdOp>(
-        rewriter.getUnknownLoc(), SmallVector<Type, 1>{}, 0, 0, memref, empty,
-        empty, empty, staticOffsets, staticSizes, staticStrides,
-        objFifo.getName(), bdIdOp.getValue(), issueToken);
-  }
-  if (dmaOp.getTarget()) {
-    auto targetLogicalObjFifo = dyn_cast<AMDAIE::LogicalObjectFifoFromMemrefOp>(
+    logicalObjFifo = dyn_cast<AMDAIE::LogicalObjectFifoFromMemrefOp>(
         dmaOp.getTarget().getDefiningOp());
-    if (!targetLogicalObjFifo) {
+    if (!logicalObjFifo) {
       return dmaOp.emitOpError() << "expected target to be an "
                                     "`amdaie.logicalobjectfifo.from_memref`";
     }
-    if (!dmaOp.hasTargetAddressing()) {
-      return dmaOp.emitOpError()
-             << "expected target addressing for DMA with target on L3";
-    }
-    AMDAIE::BdIdOp bdIdOp = dmaOp.getTargetBdIdOp();
-    if (!bdIdOp)
-      return dmaOp.emitOpError() << "expected to have a target BD ID op";
-
-    // DmaOp either has explicit target addressing OR the defining op of its
-    // source has its target on L3.
-    SmallVector<Value> empty;
-    SmallVector<int64_t, 4> staticOffsets(4, 0);
-    SmallVector<int64_t, 4> staticSizes(4, 1);
-    SmallVector<int64_t, 4> staticStrides(4, 0);
-    if (failed(getStaticDimsForExplicitAddressing(
-            dmaOp, dmaOp.getTargetMixedOffsets(), dmaOp.getTargetMixedSizes(),
-            dmaOp.getTargetMixedStrides(), staticOffsets, staticSizes,
-            staticStrides))) {
-      return failure();
-    }
-    if (failed(canonicalizeNpuStridedPatternForAIE(staticOffsets, staticSizes,
-                                                   staticStrides))) {
-      return dmaOp.emitError() << "could not canonicalize for AIE";
-    }
+  }
 
-    AMDAIE::CircularDmaCpyNdOp dmaCpyNd = dmaOp.getDmaCpyNdOp();
-    Value memref = bindingsMapper.lookup(targetLogicalObjFifo.getMemref());
-    auto objFifo = dyn_cast<xilinx::AIE::ObjectFifoCreateOp>(
-        mapper.lookup(dmaCpyNd.getOperation()));
-    if (!objFifo) {
-      return dmaOp.emitError()
-             << "input isn't mapped to an `aie.objectifo` operation";
-    }
-    bool issueToken = dmaOp.hasDmaWaitOpUser();
-    rewriter.create<AIEX::NpuDmaMemcpyNdOp>(
-        rewriter.getUnknownLoc(), SmallVector<Type, 1>{}, 0, 0, memref, empty,
-        empty, empty, staticOffsets, staticSizes, staticStrides,
-        objFifo.getName(), bdIdOp.getValue(), issueToken);
+  else {
+    return dmaOp.emitOpError()
+           << "has neither source not target memory space as L3.";
   }
+
+  Value memref = bindingsMapper.lookup(logicalObjFifo.getMemref());
+
+  auto objFifo =
+      dyn_cast<AIE::ObjectFifoCreateOp>(mapper.lookup(dmaCpyNd.getOperation()));
+
+  uint32_t bdId = bdIdOp.getValue();
+
+  if (!objFifo)
+    return dmaOp.emitError()
+           << "input isn't mapped to an `aie.objectifo` operation";
```

Comment:
```suggestion
  if (!objFifo) {
    return dmaOp.emitError()
           << "input isn't mapped to an `aie.objectifo` operation";
  }
```
https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching